File: ros/ws02/first_ws/src/use_custom_msg_pkg/src/subscriber_cus_msg.cpp
//
// Created by zcb on 5/3/21.
//

#include <iostream>
#include <ros/ros.h>
#include <custom_msgs/Student.h>


void topicCallback(const custom_msgs::Student::ConstPtr &msg) {
    std::cout << msg->name << " || " << msg->age << std::endl;
}

int main(int argc, char **argv) {

    std::string nodeName = "subscriber_node";
    ros::init(argc, argv, nodeName);
    ros::NodeHandle node;

    std::string topicName = "/zcb01/topic";
    ros::Subscriber sub = node.subscribe(topicName, 1000, topicCallback);

    ros::spin();
}

File: 06ros/ws02/first_ws/src/use_custom_msg_pkg/scripts/subscriber_cus_msg.py
#! /usr/bin/env python
# coding:utf8

import rospy
from custom_msgs.msg import Student


def topic_callback(msg):
    if not isinstance(msg, Student):
        return
    print "%s %d" % (msg.name, msg.age)


if __name__ == '__main__':
    nodeName = "subscriber_node"
    rospy.init_node(nodeName)

    topicName = "/zcb01/topic"
    rospy.Subscriber(topicName, Student, topic_callback)

    rospy.spin()
